File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/2473-max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
    def maximumSum(self, nums):
        mp = [0] * 82  
        ans = -1

        for num in nums:
            sum_digits = sum(int(d) for d in str(num))

            if mp[sum_digits]:
                ans = max(ans, num + mp[sum_digits])

            mp[sum_digits] = max(mp[sum_digits], num)

        return ans

File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/2473-max-sum-of-a-pair-with-equal-sum-of-digits.java
class Solution {
    public int maximumSum(int[] nums) {
        int[] mp = new int[82];  
        int ans = -1;

        for (int num : nums) {
            int sumDigits = digitSum(num);

            if (mp[sumDigits] > 0)
                ans = Math.max(ans, num + mp[sumDigits]);

            mp[sumDigits] = Math.max(mp[sumDigits], num);
        }
        return ans;
    }

    private int digitSum(int num) {
        int sum = 0;
        while (num > 0) {
            sum += num % 10;
            num /= 10;
        }
        return sum;
    }
}
